### PR TITLE
docs: update README.md

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Updated Ruby SDK repo readme 

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,75 +1,115 @@
...
+To use the Ruby SDK, create a client instance as shown:
 
 ```ruby
 analytics = RudderAnalyticsSync::Client.new(
-  write_key: <YOUR_WRITE_KEY>, # required
-  data_plane_url: <DATA_PLANE_URL>
+  write_key: 'WRITE_KEY', # Required
+  data_plane_url: 'DATA_PLANE_URL',
+  stub: false,
+  gzip: false,  # set to true to enable Gzip compression
```
Comment (pallabmaiti) [inline README.md]:
```suggestion
  gzip: true,  # set to false to disable Gzip compression
```

File: README.md [hunk 2]
```diff
@@ -1,75 +1,115 @@
...
+## Gzip support
+
+The Ruby SDK supports Gzip compression from version 2.0.0 and it is enabled (set to `true`) by default. However, you can disable this feature by setting the `Gzip` parameter to `false` while initializing the SDK, as shown:
+
+```ruby
+analytics = RudderAnalyticsSync::Client.new(
+  write_key: 'WRITE_KEY', # required
+  data_plane_url: 'DATA_PLANE_URL',
+  stub: false,
+  gzip: false, // set to true to enable Gzip compression
```
Comment (pallabmaiti) [inline README.md]:
```suggestion
  gzip: true, // set to false to disable Gzip compression
```

File: README.md [hunk 3]
```diff
@@ -1,75 +1,115 @@
...
+  on_error: proc { |error_code, error_body, exception, response|
+    # defaults to an empty proc
+  }
+)
+```
+
+| Note: Gzip requires `rudder-server` version 1.4 or later. |
```
Comment (pallabmaiti) [inline README.md]: Can you bring this to top & if possible more eye-catching way?

**Review by pallabmaiti — CHANGES_REQUESTED** [review]

**Review by pallabmaiti — APPROVED** [review]

**Comment by ameypv-rudder** [comment]
Closing this PR as a fresh PR is raised to avoid merge conflicts.

**Pull request closed without merging**